### PR TITLE
Adding v4 Support - joshua/bitly-v4-support

### PR DESCRIPTION
**Pull request by joshuatz**
## Summary of changes
 - I'm not going to list everything (this is a large rewrite after all), but *in addition* to the rewrite to make this compatible with V4 of the Bitly API, I also made some other changes
	 - Swapped `request` with `axios`
	 - Swapped `sepia` for `replay` (lets me customize headers and avoid saving auth token to checked-in files)
	 - Updated type defs, added type guards
	 - Updated TypeDoc JSDoc comments
	 - Updated README and inline documentation

## Misc notes
 - Skipped test
	 - The endpoint to get clicks is currently failing if the requested time unit is `minute` - it spits back a 500 internal error. The exact same request was working just days ago.
	 - I reached out to Bitly, but have not heard back yet.
	 - For now, [I'm just skipping the test](https://github.com/joshuatz/node-bitly/blob/a6d4157f18d2e785e7e0740bcca883f22931330f/src/bitly.spec.ts#L136-L137), and added a new test for the `day` unit, to replace it

## Error Handling
Rather than having request promises return a union of valid response and error (something like `Promise<BitlyResponse|BitlyError>`), I opted to re-throw errors generated by Bitly, but I also added some helpful type guards to help with type narrowing on exceptions (e.g. , in the case of Bitly API error vs your internet is unplugged). Basically, the dev using `node-bitly` might have to type check in the `catch {}` block, instead of needing to check for response-vs-error in the `try {}` block.

This is a personal preference, up for discussion. Relevant work is in commit 3ceed9d67d1041075c436475208e060fffb4a5ff.

## Version Bumping and Backward Compatibility
It looks like you were maybe planning on adding V4 compatability as a "minor" release patch. I'm going off this text in the readme:

> Version 6 is the current in-development version, re-written in Typescript. This version currently only supports the **Bitly v3** API and will continue to do so in `v6.0.x`. Version `6.1.x` will introduce **Bitly v4** support.

Since it is currently at v6.1.0, the new version would be 6.2.0 if doing a minor bump. However, one of the hallmarks of a minor bump is that it should be backwards compatible with the previous releases, and that poses a problem here.

So far, node-bitly has been pretty much passing back the response from the Bitly API to the node-bitly consumer "as-is". This means, that in order to provide backwards compatibility, I would essentially have to transform and wrap all the Bitly V4 responses, to make them "look" like V3 responses. For example, let's consider the main "shorten" endpoint:

This V4 response (2020):
```json
{
    "created_at": "2020-02-03T12:06:17+0000",
    "id": "bit.ly/38XaXKy",
    "link": "http://bit.ly/38XaXKy",
    "custom_bitlinks": [],
    "long_url": "https://github.com/tanepiper/node-bitly",
    "archived": false,
    "tags": [],
    "deeplinks": [],
    "references": {
        "group": "https://api-ssl.bitly.com/v4/groups/8sBebUiuV1s"
    }
}
```
... would need to get mocked as / transformed into this V3 response:
```json
{
    "status_code": 200,
    "status_txt": "OK",
    "data": {
        "url": "http://bit.ly/38XaXKy",
        "hash": "38XaXKy",
        "global_hash": "bT5Rke",
        "long_url": "https://github.com/tanepiper/node-bitly",
        "new_hash": 0
    }
}
```

I haven't started going down this route, because frankly it feels kludgy, and part of it is straight up not possible (just one example: the Bitly API no longer returns a `global_hash`, so I can't return it to mock a V3 response).

### TLDR:
***TLDR***: I think it makes more sense to just release this a brand new major release - **v7.0.0** - and have instructions to developers on how to update their use of node-bitly to handle the new V4 responses. My PR reflects that decision, but I am happy to discuss alternatives.

# Wrap-up
Let me know if you have any questions, desired changes, etc. As far as functionality goes, I believe this should be ready-to-roll.

The only major things that I see as being left are the specifics of rolling out a new release - adding a git tag, changing the default branch in your repo to v7.x.x (and merging this into it), re-generating docs, and publishing on NPM.

### CODE REVIEW

**Comment by tanepiper** [comment]
@joshuatz Wow - I won't have time to fully review it until later this week, possibly the weekend but I'll aim to get it merged ASAP and released.

Fantastic work!

**Comment by joshuatz** [comment]
@tanepiper Thanks! Feel free to ping me with any questions, feedback, etc. while reviewing!

**Comment by tanepiper** [comment]
@joshuatz Hey just wanted to give an update that I didn't get time this weekend to review this, I'll try make it a priority for this week!

**Comment by joshuatz** [comment]
@tanepiper No worries; appreciate the update!
